### DISCUSSION TITLE
Tshock is starting as a vanilla server?

### DISCUSSION THREAD

**Discussion by crazygmr101** [Pryaxis/TShock#2282] [Q&A] [answered]
So.. here's an.. odd question. I followed the instructions for downloading TShock found [here](https://tshock.readme.io/docs), but, when I start the server with `mono TerrariaServer.exe` is just.. looks like a vanilla Terraria server.

![image](https://user-images.githubusercontent.com/22755628/115328985-8b8a6500-a15f-11eb-9bc7-7aa319023356.png)

![image](https://user-images.githubusercontent.com/22755628/115328990-8f1dec00-a15f-11eb-887f-c5966f50a729.png)

I even `tee`'ed it to a file to force console to not clear, so I didn't miss anything, and yea, it just looks to be vanilla.

![image](https://user-images.githubusercontent.com/22755628/115329040-abba2400-a15f-11eb-92cd-2751f3c650d2.png)
 
(IP is my public-facing proxy, i didn't forget to censor :p)

**EDIT** - The above post was for 4.5.1, 4.5.0.1 seems to work fine.

![image](https://user-images.githubusercontent.com/22755628/115329364-33a02e00-a160-11eb-8837-b7c353e45b70.png)

**Comment by hakusaro** [comment, marked as answer]
This is because TShock is a plugin for Terraria Server API. We just so happen to distribute TSAPI with TShock. The `TerrariaAPI version` line indicates that you're on TSAPI `2.1.0.0` for Terraria protocol `1.4.2.1` (which is internally called release `236`), on Open Terraria API version `1.4.2.1` (so, that's TSAPI on OTAPI).

In `ServerPlugins` you need to have a `TShockAPI.dll` file, which is actually TShock. While we'd love to rename that so it doesn't include `API`, we can't without breaking out extremely simple upgrade instructions, which involve just copying the new release onto the old release. For posterity, the `API` in `TShockAPI` refers to the fact that this is the version of `TShock` built for `TSAPI`.

So, tl;dr: your issue is solved, but in the future, if you don't see TShock initializing, check `ServerPlugins` for `TShockAPI.dll`.
